lib/process-services-cloud/src/lib/group/services/identity-group.service.ts

```diff
@@ -0,0 +1,110 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Injectable } from '@angular/core';
+import { AppConfigService, OAuth2Service } from '@alfresco/adf-core';
+import { EMPTY, Observable, throwError } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { IdentityGroupServiceInterface } from './identity-group.service.interface';
+import { IdentityGroupFilterInterface } from './identity-group-filter.interface';
+import { IdentityGroupModel } from '../models/identity-group.model';
+
+@Injectable({ providedIn: 'root' })
+export class IdentityGroupService implements IdentityGroupServiceInterface {
+
+    context: string = '';
+    queryParams: { search: string; application?: string; roles?: string [] };
+
+    constructor(
+        private oAuth2Service: OAuth2Service,
+        private appConfigService: AppConfigService
+    ) {}
+
+    public search(name: string, filters?: IdentityGroupFilterInterface): Observable<IdentityGroupModel[]> {
+        if (name.trim() === '') {
+            return EMPTY;
+        } else if (filters?.withinApplication !== undefined && filters?.withinApplication !== '') {
```

Comment:
`filters?.withinApplication` shoudn't be enough? 
[ECMA docs](https://262.ecma-international.org/5.1/#sec-9.2)